SampleGallery/Samples/SDK 16299/LinearGradients/LinearGradients.xaml.cs

```diff
@@ -0,0 +1,257 @@
+            _changeb1stop1.Duration = TimeSpan.FromSeconds(2);
+
+            _changeb1stop2 = _compositor.CreateColorKeyFrameAnimation();
+            _changeb1stop2.InsertKeyFrame(.5f, Colors.Teal);
+            _changeb1stop2.Duration = TimeSpan.FromSeconds(2);
+        }
+
+        /*
+         * Run animation on target visual brush
+         */
+
+        private void Pointer_Pressed(object sender, RoutedEventArgs e)
+        {
+            pointerPressedChangeColors(b1GradientStop1, b1GradientStop2);
+        }
+
+        private void pointerPressedChangeColors(CompositionColorGradientStop target, CompositionColorGradientStop target1)
+        {
+
+            if (target.Color == Colors.DeepPink || target.Color == Colors.Honeydew)
+            {
+                target.StartAnimation(nameof(target.Color), _changeb1stop1);
+                target1.StartAnimation(nameof(target1.Color), _changeb1stop2);
+            }
+            else if (target.Color == Colors.LightSkyBlue || target.Color == Colors.Teal)
+            {
+                target1.StartAnimation(nameof(target1.Color), _colorAnimBrush1Stop1);
+                target.StartAnimation(nameof(target.Color), _colorAnimBrush1Stop2);
+            }
+        }
+
+        private void AnimateGradient(SpriteVisual target)
+        {
+            target.StartAnimation("Scale", _scaleAnim);
+        }
+
+        private void AnimateBrushStop1(CompositionColorGradientStop target, CompositionColorGradientStop target1)
+        {
+            if (target.Color == Colors.DeepPink)
+            {
+                target.StartAnimation(nameof(target.Color), _colorAnimBrush1Stop1);
+                target1.StartAnimation(nameof(target1.Color), _colorAnimBrush1Stop2);
+            } else if (target.Color == Colors.Teal)
+            {
+                target.StartAnimation(nameof(target.Color), _changeb1stop1);
+                target1.StartAnimation(nameof(target1.Color), _changeb1stop2);
+            }
+        }
+
+        private void ChangeBrushBack(CompositionColorGradientStop target, CompositionColorGradientStop target1)
+        {
+            if (target.Color == Colors.Honeydew)
+            {
+                target1.StartAnimation(nameof(target1.Color), _colorAnimBrush1Stop1);
+                target.StartAnimation(nameof(target.Color), _colorAnimBrush1Stop2);
+            } else if (target.Color == Colors.LightSkyBlue)
+            {
+                target1.StartAnimation(nameof(target1.Color), _changeb1stop1);
+                target.StartAnimation(nameof(target.Color), _changeb1stop2);
+            }
+        }
+
+        private void AnimateOffset(SpriteVisual target)
+        {
+            float endPoint = target.Size.X;
+            float startPoint = target.Size.X - target.Size.X;
+
+            if (target.Offset.X == startPoint)
+            {
+                target.AnchorPoint = new Vector2(1f, 0f);
+                _offsetAnim.InsertKeyFrame(1, endPoint);
```

Comment:
While technically correct, it's kind of weird that this method has a side effect of modifying the keyframes every time it's called.  Since that animation isn't used outside of this method, perhaps we should simply create it on demand here rather than storing it as a member variable?

Reply:
Also, unless I'm missing something, I think this method is over-complicated.
_offsetAnim is created with no keyframes, and this method updates the final keyframe to be either:
"offset == target's size, using an anchor point of target's width"
Or
"offset == 0, using an anchor point of 0"

In either of those two cases, the left edge of the Visual will end up at offset 0, so I think you can simply just create that animation up front and always use it, without mucking with the AnchorPoint at all?